### ISSUE TITLE
Remove redundant per-variant normalization in CV notebooks

### ISSUE THREAD

**Issue by jgallowa07** [matsengrp/multidms#231]
# Remove redundant per-variant normalization in CV notebooks

## Summary

Following the merge of PR #230 (closing #227), `functional_score_loss` now uses `.mean()` instead of `.sum()`, so `total_loss_training` and `total_loss_validation` on `ModelCollection.fit_models` are already per-variant averages. The cross-validation notebooks in both experiment pipelines still divide those columns by sample count before plotting, producing doubly-normalized values. This task removes that dead normalization while preserving the on-disk CSV schema so downstream figure code continues to work unchanged.

## Motivation

PR #230 changed the loss scale at the source. The CV notebooks were authored when loss was a total (summed) quantity, where dividing by variant count was the right way to make training and validation losses visually comparable. After the merge, that division silently compounds the normalization — the plotted y-axis still represents something meaningful (loss / n_samples²-ish), but it no longer matches the claim in the axis label ("mean Huber loss per variant"), and the cross-pipeline comparison with the V0.4.0-anchored hyperparameter grids becomes hard to interpret.

If we don't fix this, manuscript figures and the spike CV diagnostic plots will continue to report misleading values.

Related: PR #230, issue #227.

### In scope

- Remove the `n_samples` lookup dict and the `cv_data["mean_loss"] = cv_data["loss"] / cv_data["n_samples"]` division in both CV notebooks.
- Keep `mean_loss` as a column in the emitted `cross_validation_loss.csv` — as an alias for `loss` — so downstream consumers (notably `manuscript_figures.ipynb`) do not need schema changes in this task.
- Delete the stale `mean_loss` profile from the scv2-spike pipeline: remove `experiments/scv2-spike/config/config_mean_loss.yaml` and the corresponding `profile == "mean_loss"` branch in `experiments/scv2-spike/Snakefile`. The default `config.yaml` was already updated with the rescaled V0.4.0-anchored parameter grid, so the separate profile is redundant.
- Re-run the `test` profile of both pipelines locally to verify notebook execution and sanity-check the plotted values.
- Leave the y-axis label "mean Huber loss per variant" as-is (still accurate, now sourced directly from the model).

### Out of scope

- Running the `experimental` or `prod` profile of either pipeline (can happen in a follow-up).
- Refactoring `manuscript_figures.ipynb` to read `loss` instead of `mean_loss`. The alias column preserves its current reader; a cleanup pass can drop the alias later.
- Touching `evaluate.ipynb` or `fit_models.ipynb` in either pipeline — a grep sweep confirmed they don't consume `total_loss_*` or `mean_loss` for plotting.
- Updating `experiments/dashboard.py` — grep confirmed it does not reference these columns.
- Updating `experiments/loss-normalization/` notebooks — they don't use CV outputs.
- Re-running already-persisted result CSVs on disk. Users who want updated values re-run the relevant pipeline.

### 1. Simulation CV notebook

**File:** `experiments/simulation/notebooks/cross_validation.ipynb` (existing)

Target the cell that starts with `cv_data = mc_cv.fit_models.copy()` and builds `n_samples`. Replace the dict construction and the redundant division with a direct alias.

```python
# Before:
n_samples = {}
for d in train_datasets:
    n_train_variants = sum(
        len(d.variants_df.query(f"condition == '{c}'"))
        for c in d.conditions
    )
    n_samples[(d.name, "training")] = n_train_variants
for name, test_df in filtered_test.items():
    n_test_variants = sum(
        len(test_df.query(f"condition == '{c}'"))
        for c in test_df["condition"].unique()
    )
    n_samples[(name, "validation")] = n_test_variants

cv_data["n_samples"] = cv_data.apply(
    lambda row: n_samples.get(
        (row["dataset_name"], row["dataset"]), 1
    ),
    axis=1,
)
cv_data["mean_loss"] = cv_data["loss"] / cv_data["n_samples"]

# After:
# total_loss_{training,validation} are already per-variant averages
# (functional_score_loss uses .mean() as of PR #230).
# Keep `mean_loss` as an alias for downstream CSV consumers.
cv_data["mean_loss"] = cv_data["loss"]
```

The downstream plotting block that references `cv_data["mean_loss"]` stays unchanged.

### 2. Spike CV notebook

**File:** `experiments/scv2-spike/notebooks/cross_validation.ipynb` (existing)

Same transformation as above — target the cell that builds `n_samples` and computes `cv_data["mean_loss"] = cv_data["loss"] / cv_data["n_samples"]`. Remove the dict construction and the division; alias `mean_loss = loss`.

Note: the simulation pipeline's `Snakefile` is untouched — it has no `mean_loss` profile branch to remove. Only the spike `Snakefile` needs editing (§4 below).

### 3. Manuscript figures notebook

**File:** `experiments/simulation/notebooks/manuscript_figures.ipynb` (existing)

No change required. It reads `mean_loss` from `cross_validation_loss.csv` (line ~374: `loss_col = "mean_loss" if "mean_loss" in cv_sub.columns else "loss"`). Since we preserve the `mean_loss` alias column, this notebook picks up the corrected values automatically.

### 4. Remove stale `mean_loss` profile from spike pipeline

**Files:**
- `experiments/scv2-spike/config/config_mean_loss.yaml` (delete)
- `experiments/scv2-spike/Snakefile` (edit)

The default `config.yaml` has already absorbed the V0.4.0-anchored parameter grid (rescaled `fusionreg`, `l2reg`, `beta0_ridge`, `lasso_choice=4.0e-5`). `config_mean_loss.yaml` was a transitional profile that is now a confusing duplicate — it differs from default only in `output_dir: results-mean-loss` and `skip_cross_validation: true`.

Snakefile changes (`experiments/scv2-spike/Snakefile`, in the profile-dispatch block near the top):

```python
# Before:
if profile not in (None, "test", "experimental", "mean_loss"):
    raise ValueError(
        f"Unknown profile '{profile}'. Use 'test', 'experimental', 'mean_loss', or leave unset for production."
    )
...
elif profile == "mean_loss":
    CONFIG_PATH = os.path.join(SPIKE_DIR, "config", "config_mean_loss.yaml")

# After:
if profile not in (None, "test", "experimental"):
    raise ValueError(
        f"Unknown profile '{profile}'. Use 'test', 'experimental', or leave unset for production."
    )
# (drop the `elif profile == "mean_loss"` branch entirely)
```

## Migration / Rollback Plan

No migration required — the CSV schema is preserved (same columns, same dtypes). Any previously-persisted `cross_validation_loss.csv` on disk remains readable but contains stale doubly-normalized values; users should re-run the relevant pipeline to regenerate.

Any `results-mean-loss/` directories on disk (the `output_dir` written by the deleted profile) are intentionally left alone — cleanup of old result trees is out of scope.

**Rollback:** revert the notebook edits and restore `config_mean_loss.yaml` plus the Snakefile branch. No data or state to restore.

## Testing Strategy

- Run the `test` profile of both pipelines locally, using the invocation documented in each Snakefile's header (`snakemake -s experiments/simulation/Snakefile --config profile=test -j4` and the analogous spike command).
- Confirm both pipelines complete without error.
- Spot-check `cross_validation_loss.csv`: the `loss` and `mean_loss` columns should be exactly equal, and values should sit on the scale of `total_loss_training` recorded in `fit_models.pkl` (i.e., a per-variant Huber-loss magnitude, not divided again).
- **Correctness validation against the fitter:** at one fusionreg grid point, read the corresponding fitted model's convergence trajectory and compare the plotted CV `mean_loss` against `loss_per_variant_total` from the final trajectory step. These should agree up to the Huber parameter and train/test split definition — a doubly-normalized plot would be roughly `n_variants`× smaller, which is easy to spot.
- Verify the CV plot in each notebook renders and the y-axis values are in the expected per-variant range.
- No new unit tests — this is a notebook-only change and the underlying loss change is already tested by `TestMeanLossNormalization` in `tests/test_jaxmodels.py`.

## Dependencies

No new dependencies.

## Acceptance Criteria

- [ ] `experiments/simulation/notebooks/cross_validation.ipynb` no longer constructs `n_samples` or divides `loss` by it.
- [ ] `experiments/scv2-spike/notebooks/cross_validation.ipynb` no longer constructs `n_samples` or divides `loss` by it.
- [ ] Both CV notebooks still emit a `cross_validation_loss.csv` with columns including `loss` and `mean_loss`, where `mean_loss == loss` row-wise.
- [ ] `manuscript_figures.ipynb` runs end-to-end against the updated CSV and produces a CV-loss panel with per-variant-scale values.
- [ ] `experiments/scv2-spike/config/config_mean_loss.yaml` is deleted and the Snakefile no longer accepts the `mean_loss` profile.
- [ ] `test` profile of both pipelines completes cleanly locally using the default `config.yaml` path.
- [ ] `ruff check .` and `black --check .` pass (notebook cells included via the existing CI setup).

## Enhancements to CLAUDE.md

Consider adding a note under the "Key design patterns" section reinforcing:

> Since PR #230, `ModelCollection.fit_models["total_loss_*"]` columns are per-variant averages (one per condition, plus `"total"` averaged over conditions). Do not divide again by variant count when plotting — it will produce doubly-normalized values.
